data_portal_api.yml

```diff
@@ -0,0 +1,334 @@
+openapi: 3.0.0
+info:
+  version: "1.0.0"
+  title: Copora Data Portal API
+  description: >
+    This is the API for the Copora Data Portal
+servers:
+  - description: production
+    url: https://copora.czi-single-cell.science
+paths:
+  /v1/submissions:
+    get:
+      summary: List submissions
+      tags:
+        - submissions
+      description: >-
+        This lists all submissions and their UUIDs that currently exist in corpora. If a parameter is specified as a filter, then only projects that meet the status criteria will be outputted.
+
+        If user uuid is provided, submissions from that user are returned. If no user uuid is specified the id of the authenticated user is used.
+      parameters:
+        - name: status
+          description: The status of the submissions to list.
+          in: query
+          schema:
+            type: string
+            enum: [all, open, closed]
```

Comment:
Does there need to be an "all" here? All is already handled when the parameter is not specified at all.

Reply:
Also might be helpful to have a description that is more tailored to the API call itself so that we can distinguish between, say, a submission not found 202 and a project not found 404.

Reply:
we can add details in the error response. Adding a different response code isn't the right solution IMO.

Reply:
SG